pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -103,27 +103,34 @@ func (l *PodInterface) PlugPhase1(vmi *v1.VirtualMachineInstance, iface *v1.Inte
 	if !isExist {
 		err := driver.discoverPodNetworkInterface()
 		if err != nil {
-			return err
+			return createCriticalNetworkError(err)
 		}
 
 		if err := driver.preparePodNetworkInterfaces(); err != nil {
-			log.Log.Reason(err).Critical("failed to prepare pod networking")
+			log.Log.Reason(err).Error("failed to prepare pod networking")
+			return createCriticalNetworkError(err)
```

Comment:
Could you please explain why a simple `fmt.Errorf` is not enough?
I do not understand the need for the special struct, its method and this function.

If this is just for asserting against a specific error, then isn't `errors.New` more appropriate?

Reply:
This struct is needed for the virt-handler controller to differentiate between regular error to critical error. If `phase1` failed with regular error, the controller will re-enqueue the vmi and on the next cycle of the controller loop another attempt to invoke `phase1` will be performed.

In case  `phase1` failed with critical error (unrecoverable error). The controller we move the vmi to `Failed` phase and the vmi won't be re-enqueued.